compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2232,24 +2233,49 @@ public void visit(BLangTableConstructorExpr tableConstructorExpr, AnalyzerData d
 
     @Override
     public void visit(BLangRecordLiteral recordLiteral, AnalyzerData data) {
+        BType referredType = Types.getReferredType(recordLiteral.getBType());
         List<RecordLiteralNode.RecordField> fields = recordLiteral.fields;
 
         for (RecordLiteralNode.RecordField field : fields) {
             if (field.isKeyValueField()) {
                 analyzeExpr(((BLangRecordKeyValueField) field).valueExpr, data);
+                reportIfDeprecatedUsage(((BLangRecordKeyValueField) field).key.fieldSymbol, recordLiteral,
+                        ((BLangRecordKeyValueField) field).pos);
             } else if (field.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
-                analyzeExpr((BLangRecordLiteral.BLangRecordVarNameField) field, data);
+                BLangRecordLiteral.BLangRecordVarNameField recField
+                        = (BLangRecordLiteral.BLangRecordVarNameField) field;
+                analyzeExpr(recField, data);
+
+                if (referredType.getKind() == TypeKind.RECORD) {
```

Comment:
Let's use type tag instead of kind. Fix everywhere.

Reply:
Sure. However, what is the reason for it?

Reply:
IMO, it's used more often and therefore, a greater guarantee for correctness, but it is also just an int comparison.